bigtable-hbase-replication/src/test/java/com/google/cloud/bigtable/hbase/replication/HbaseToCloudBigtableReplicationEndpointTest.java

```diff
@@ -204,23 +200,49 @@ public void testDelete() throws IOException, InterruptedException {
 
 
     int i =1; // Row 0 has been deleted.
-    for(Result result: cbtTable.getScanner(new Scan())){
+    assertResultRowsInCBT(cbtTable, 5, 1);
+  }
+
+  //TODO create a multi column family
+  @Test
+  public void testMultiTableReplication() throws IOException, InterruptedException {
+    Table table1 = hbaseTestingUtil.getConnection().getTable(TableName.valueOf(TABLE_NAME));
+    Table table2 = hbaseTestingUtil.getConnection().getTable(TableName.valueOf(TABLE_NAME_2));
+    for (int i = 0; i < 10; i++) {
+      String rowKey = ROW_KEY_PREFIX + i;
+      Put put = new Put(rowKey.getBytes());
+      byte[] val = (VALUE_PREFIX + i).getBytes();
+      put.addColumn(CF1, COL_QUALIFIER, 0, val);
+      table1.put(put);
+      table2.put(put);
+    }
+
+    // Wait for replication to catch up
+    // TODO Find a better alternative than sleeping? Maybe disable replication or turnoff mini cluster
+    Thread.sleep(2000);
+    //   Waiter.waitFor(CONF, 60000, () -> TestEndpoint.getEntries().size() >= cellNum);
+
+    Table cbtTable = cbtConnection.getTable(TableName.valueOf(TABLE_NAME));
+    Table cbtTable2 = cbtConnection.getTable(TableName.valueOf(TABLE_NAME_2));
+    assertResultRowsInCBT(cbtTable, 10, 0);
```

Comment:
This does not test that the replication happens in parallel? 

We need to test that the replication is happening in parallel. Create a custom TestEndpoint and have some mutexes in it. Or leave hooks in beginning and ending to add some counters to make sure that tables were replicating parallely.

Reply:
I am creating a custom TestEndpoint for this test.

Basically

```class HbaseToCloudBigtableReplicationEndpointForTest extends HbaseToCloudBigtableReplicationEndpoint``` 

The parallel replication is abstracted under `replicate` method under `replicationContext.getEntries`. Can I override the entire `replicate` method for test? Any suggestions.

Reply:
You will need to create hooks where you can inject test code. One way to do it maybe to supply TestTableReplciators and using them to book keep. If you replicate a WAL entry with 3 tables, you could then expect the test tableReplicators to increment a static atomic integer and block untill it reaches 3. when it reaches 3 you complete the future and all 3 tableReplicators will be free.